### PR TITLE
minor refactor for bishop and rook

### PR DESCRIPTION
**Pull request by johnellison**
Nothing major here. Just taking out unnecessary `self`, `do` and a few other things. Reduce code bishop and rook move logic by 18 lines. Just trying to keep our code lean.

### CODE REVIEW

File: app/models/rook.rb [hunk 1]
```diff
@@ -19,39 +14,30 @@ def legal_move?(x, y)
   # of any obstruction will return false.
 
   def obstructed_move?(x, y)
-    pos_x = self.x_position
-    pos_y = self.y_position
-    game = self.game
+    pos_x = x_position
+    pos_y = y_position
```
Comment (washburnad) [inline app/models/rook.rb]: great catch.  I forget that you don't have to explicitly call self.  And it completely eliminates line 24.

**Comment by washburnad** [comment]
Awesome.

**Comment by washburnad** [comment]
Once again, can't merge automatically, though.

**Comment by johnellison** [comment]
I wonder why that keeps happening. Want me to merge using the command line?

**Comment by washburnad** [comment]
Yeah, go ahead and try.  The other option, I think, is to go back to your local copy and merge the most recent origin/master into the branch.  Then push it already merged.

**Comment by johnellison** [comment]
That makes sense. I hadn't run into this issue yet. Thanks. I'll merge master into this branch and then push.

**Pull request merged**